### DISCUSSION TITLE
What is the best way to use a SignalR service for my Blazor RenderMode.InteractiveServer app?

### DISCUSSION THREAD

**Discussion by DavidThielen** [dotnet/aspnetcore#56597] [Q&A] [answered]
Hi all;

I've read in numerous places that having my Blazor server use a SignalR service will let it handle a lot more simultaneous requests. So I want to add this. I've seen three very different ways describing how to do this:
1. [Write a ton of code](https://learn.microsoft.com/en-us/answers/questions/1724514/use-signalr-service-with-my-blazor-server-app-runn?page=1#answer-1555870)
2. [Create a SignalR app](https://github.com/aspnet/AzureSignalR-samples/tree/main/samples/ServerSideBlazor)
3. [Connect to a SignalR service](https://blazorhelpwebsite.com/ViewBlogPost/32)
4. Another approach???

I like option 3 best because it seems like a very clean way to accomplish this. But that does not mean it's the optimum approach.

So which is the optimum approach?

thanks - dave

**Comment by leetheman** [comment]
I am curious about this also - would be keen to get some official guidance

**Comment by DavidThielen** [comment, marked as answer]
According to Microsoft support - [best approach is (3) above.](https://learn.microsoft.com/en-us/answers/questions/1805570/what-is-the-best-way-to-use-a-signalr-service-for)
